core/src/consensus/mod.rs

```diff
@@ -44,27 +44,48 @@ use ctypes::util::unexpected::{Mismatch, OutOfBounds};
 use primitives::{Bytes, H256, U256};
 
 use self::epoch::{EpochVerifier, NoOp, PendingTransition};
+use self::tendermint::types::View;
 use crate::account_provider::AccountProvider;
 use crate::block::SealedBlock;
 use crate::codechain_machine::CodeChainMachine;
+use crate::encoded;
 use crate::error::Error;
 use crate::header::Header;
 use crate::parcel::{SignedParcel, UnverifiedParcel};
 use crate::scheme::CommonParams;
 use crate::views::HeaderView;
 use Client;
 
-/// Seal type.
-#[derive(Debug, PartialEq, Eq)]
 pub enum Seal {
-    /// Proposal seal; should be broadcasted, but not inserted into blockchain.
-    Proposal(Vec<Bytes>),
-    /// Regular block seal; should be part of the blockchain.
-    Regular(Vec<Bytes>),
-    /// Engine does generate seal for this block right now.
+    Solo,
+    SimplePoA(Signature),
+    Tendermint {
+        prev_view: View,
+        cur_view: View,
+        signatures: Vec<Signature>,
+    },
     None,
 }
 
+impl Seal {
+    pub fn seal_fields(&self) -> Option<Vec<Bytes>> {
+        match self {
+            Seal::None => None,
+            Seal::Solo => Some(Vec::new()),
+            Seal::SimplePoA(signature) => Some(vec![::rlp::encode(signature).into_vec()]),
+            Seal::Tendermint {
+                prev_view,
+                cur_view,
+                signatures,
+            } => Some(vec![
+                ::rlp::encode(cur_view).into_vec(),
```

Comment:
It made me confused.
How about swapping the order of `cur_view` and `prev_view`?

Reply:
Good